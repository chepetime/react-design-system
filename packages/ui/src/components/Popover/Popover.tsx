import React from "react";
import classnames from "classnames";
import css from "./Popover.module.scss";

export interface PopoverProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Popover = React.forwardRef<HTMLDivElement, PopoverProps>(
  (props: PopoverProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Popover)} ref={ref}>
        <p>Popover component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Popover.displayName = "Popover";
