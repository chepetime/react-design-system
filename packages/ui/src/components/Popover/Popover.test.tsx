
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Popover } from "./Popover";
import { render, screen } from "@testing-library/react";

describe("Popover", () => {
  it("renders without errors", () => {
    render(<Popover />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Popover>Some children</Popover>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Popover>Some children</Popover>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Popover />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
