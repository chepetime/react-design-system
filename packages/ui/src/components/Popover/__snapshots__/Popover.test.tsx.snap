// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover matches snapshot 1`] = `
<div>
  <div
    class="Popover"
  >
    <p>
      Popover component working!
    </p>
    <div />
  </div>
</div>
`;
